ChiModules/KEigenvalueSolver/IterativeOperations/compute_production.cc

```diff
@@ -44,10 +44,10 @@ double KEigenvalue::Solver::ComputeProduction()
         {
           double nu_sigma_f = 0.0;
           if (options.use_precursors)
-            nu_sigma_f = xs->nu_p_sigma_fg[g]+
-                         xs->nu_d_sigma_fg[g];
+            nu_sigma_f = xs->nu_prompt_sigma_f[g] +
```

Comment:
I looked at SetKsource and found this:


```script
if (options.use_precursors)                  
  fission_g += xs-&gt;chi_g[g]*xs->;
  nu_p_sigma_fg[gprime]* phi_prevp[gprime]/k_eff;                
else                  
  fission_g += xs->;chi_g[g]*xs-&gt;
  nu_sigma_fg[gprime]* phi_prevp[gprime]/k_eff;
```



I think for steady-state runs, the chi that is used should be different than the chi-prompt. For that code, it looks like what is called chi is always chi-prompt. For SS runs, chi should be the wieghted sum of chi*nu (prompt/delayed) divided by nu_total.

Reply:
In general, I second this comment, although it may be more suited for a separate issue. Nevertheless, the changes in this pull-request have highlighted inconsistencies with the total/prompt/delayed production parameters.

Reply:
We identified this before as it was quite confusing. I've created issue #146 to regear this as opposed to the renaming scope of this PR.

Reply:
> > > why is everything equal to sigma_t. I must be missing something.
> > 
> > 
> > Yup you are missing something. `sigma_t` is the primary cross-section and whenever the cross sections are made/remade we first start by clearing `sigma_t` and assigning it an all-zero value. It is then easy to mimic `sigma_t`'s structure and value to all other cross sections before we assign/read values into them.
> > To see the usefulness, compare the number of lines of code in the following two snippets:
> > ```c++
> > sigma_t.clear();
> > sigma_t.assign(num_groups, 0.0);
> > sigma_f = sigma_t;
> > sigma_a = sigma_t;
> > chi = sigma_t;
> > nu = sigma_t;
> > nu_prompt = sigma_t;
> > nu_delayed = sigma_t;
> > nu_sigma_f = sigma_t;
> > nu_prompt_sigma_f = sigma_t;
> > nu_delayed_sigma_f = sigma_t;
> > inv_velocity = sigma_t;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > vs
> > ```c++
> > sigma_t.clear();
> > sigma_t.assign(num_groups, 0.0);
> > sigma_f.clear();
> > sigma_f.assign(num_groups, 0.0);
> > sigma_a.clear();
> > sigma_a.assign(num_groups, 0.0);
> > chi.clear();
> > chi.assign(num_groups, 0.0);
> > 
> > // etc.
> > ```
> 
> Assigning sigma_t to everything seems confusing to me as well. If the goal is to simply reduce the number of source lines, why not just use assign (there's no reason to use clear()):
> 
> sigma_t.assign(num_groups, 0.0);
> sigma_f.assign(num_groups, 0.0);
> sigma_a.assign(num_groups, 0.0);
> chi.assign(num_groups, 0.0);
> .
> .
> .
> etc.

This is a better way. I implemented this.